File: completed/Progetto_pao1718/Progetto_gui/screen.cpp
#include "screen.h"
#include<iostream>


Screen :: Screen (int w, int h, double dim) : resWidth(w), resHeight(h), dimension(dim) {}

void Screen::setRes(int w, int h){
    if(w>=0 && h>=0){
    resWidth=w;
    resHeight=h;
    }
    else std::cerr<<"Input error, inserire valori >=0"<<std::endl;
}

void Screen::setDim(double dim){
    if(dim>=0)
        dimension=dim;
    else std::cerr<<"Input error, inserire valore >=0"<<std::endl;
}

QString Screen::getPrint() const{
    QString resW=QString::number(resWidth);
    QString resH=QString::number(resHeight);
    QString dim=QString::number(dimension);
    return ("Risoluzione schermo= "+resW+" X "+resH+" pixels"+'\n'+"Dimensione schermo= "+dim+" pollici"+'\n');
}

int Screen::getResW() const { return resWidth;}

int Screen::getResH() const { return resHeight;}

int Screen::getRes() const { return resHeight+resWidth;}


double Screen::getDim() const { return dimension;}

double Screen::Score() const {  return getRes()/dimension;}  //piu' e' grande lo schermo a parita' di pixel, meno alto e' lo score

double Screen::getScore() const {  return Score();}

File: completed/Progetto_pao1718/Java/esecuzione/Screen.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
//package esecuzione;

/**
 *
 * @author mhetac
 */
public class Screen implements Component {
    private double pscore() {
        if(dimension>0){
            //System.out.println("Screen::pscore()  res=" +getRes());
            //System.out.println("Screen::pscore()  dimension(int)=" +(int)dimension);
            return getRes()/dimension;
        
        }
        else {
            System.err.println("Errore: tentativo di ottenere score con schermo con dimensione <=0");
            return 0;
        }
    }
    private int resWidth;  //pixels
	    private int resHeight;  //pixels
	    private double dimension;  //diagonal inches
	    
	    public void setRes (int w, int h){  //w= larghezza   h= altezza 
	    	if(w>=0 && h>=0){
	    	    resWidth=w;
	    	    resHeight=h;
	    	    }
	    	    else System.err.println("Input error, inserire valori >=0");
	    }
	    
	    public void setDim (double dim){
	    	if(dim>=0)
	            dimension=dim;
	        else System.err.println("Input error, inserire valore >=0");
	    }
	    
	    public int getRes() {return resWidth*resHeight;}
	    public double getDim () {return dimension;}
	    public int getResW() {return resWidth;}
	    public int getResH() {return resHeight;}
	    public double getScore() {return pscore();}   //override
	    
	    public void Print() {  //override
	    	System.out.println("Risoluzione schermo= "+resWidth+" X "+resHeight);
	        System.out.println("Dimensione schermo= "+dimension);
	    }
            
            public Screen(int resW, int resH, double dim){
                resWidth=resW;
                resHeight=resH;
                dimension=dim;
            }
            
            public Screen(){
                resWidth=0;
                resHeight=0;
                dimension=0;
            }
}
